### ISSUE TITLE
Cascading errors when incorrectly extending a generic type

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#5940]
Only one error should be reported for `xyz`: "Generic type '{0}' requires {1} types argument(s).". Subsequent uses of `xyz` should not report errors. In the example below, calling `new` does, and previously referring to properties on an instance of `xyz` did too.

Note that this cleanup should not break the language services' ability to tell the user that `xyz` extends `abc` and that it will inherit members from it once a type argument is provided. (We may have to add tests to cover this case.)

From tests/cases/compiler/recursiveBaseConstructorCreation3.ts:

``` ts
    declare class base<T> {
    }
    declare class abc<T> extends base<T> {
        foo: xyz;
    }
    declare class xyz extends abc {
                              ~~~
!!! error TS2314: Generic type 'abc<T>' requires 1 type argument(s).
    }

    var bar = new xyz(); // Error: Invalid 'new' expression.
              ~~~~~~~~~
!!! error TS2351: Cannot use 'new' with an expression whose type lacks a call or construct signature.
    var r: xyz = bar.foo;
```

**Comment by sandersn** [comment]
@vladima, this tracks the follow-up to my recent bugfix #5934.

**Comment by mhegazy** [comment]
Can you elaborate on the details of the proposed fix..

**Comment by sandersn** [comment]
Once a type has failed to correctly extend a base type, do not report errors when the type or instances of the type are used. 

Unfortunately `getBaseTypes` currently logs an error (good!) but then returns an empty list as if the erroneous type just doesn't have a base type (bad!). I haven't figured out a way to change the erroneous type to `unknownType` so downstream uses will not log errors.
